naming/src/main/java/com/alibaba/nacos/naming/consistency/ephemeral/distro/DistroConsistencyServiceImpl.java

```diff
@@ -380,11 +380,14 @@ public boolean isInitialized() {
          */
         public void addTask(String datumKey, DataOperation action) {
             
-            if (services.containsKey(datumKey) && action == DataOperation.CHANGE) {
-                return;
-            }
             if (action == DataOperation.CHANGE) {
-                services.put(datumKey, StringUtils.EMPTY);
+                if (services.putIfAbsent(datumKey, StringUtils.EMPTY) != null) {
+                    // if previous change task is waiting for handling.
+                    return;
+                }
+            } else if (action == DataOperation.DELETE) {
```

Comment:
I think the change may cause lose DELETE action.

```
                } else if (action == DataOperation.DELETE) {
                    if (services.contains(datumKey)) {
                        // if new change task is waiting for handling.
                        return;
                    }
                }
```

Reply:
DELETE action will be ignored only when it's followed by a CHANGE action.

Reply:
```services``` should be separated into two groups that one group only has DELETE actions and the other only has CHANGE actions if bypass duplicated DELETE actions.